src/main/java/io/odpf/firehose/sink/bigquery/BigQuerySink.java

```diff
@@ -0,0 +1,77 @@
+package io.odpf.firehose.sink.bigquery;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.InsertAllRequest;
+import com.google.cloud.bigquery.InsertAllResponse;
+import com.google.cloud.bigquery.TableId;
+import io.odpf.firehose.consumer.Message;
+import io.odpf.firehose.exception.DeserializerException;
+import io.odpf.firehose.metrics.Instrumentation;
+import io.odpf.firehose.sink.AbstractSink;
+import io.odpf.firehose.sink.bigquery.converter.MessageRecordConverterWrapper;
+import io.odpf.firehose.sink.bigquery.handler.BigQueryRow;
+import io.odpf.firehose.sink.bigquery.models.Record;
+import io.odpf.firehose.sink.bigquery.models.Records;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BigQuerySink extends AbstractSink {
+
+    private final BigQuery bigQueryInstance;
+    private final TableId tableId;
+    private final BigQueryRow rowCreator;
+    private final Instrumentation instrumentation;
+    private final MessageRecordConverterWrapper converterWrapper;
+    private List<Message> messageList;
+
+    public BigQuerySink(Instrumentation instrumentation,
+                        String sinkType,
+                        BigQuery bigQueryInstance,
+                        TableId tableId,
+                        MessageRecordConverterWrapper converterWrapper,
+                        BigQueryRow rowCreator) {
+        super(instrumentation, sinkType);
+        this.instrumentation = instrumentation;
+        this.bigQueryInstance = bigQueryInstance;
+        this.tableId = tableId;
+        this.converterWrapper = converterWrapper;
+        this.rowCreator = rowCreator;
+    }
+
+    @Override
+    protected List<Message> execute() throws Exception {
+        Instant now = Instant.now();
+        Records records = converterWrapper.getMessageRecordConverter().convert(messageList, now);
+        InsertAllResponse response = insertIntoBQ(records.getValidRecords());
+        //parse the response.
```

Comment:
i wonder, wether its better to make more explicit by having this line extracted as method like `getInvalidMessages` instead of comments, since kind of have that intention, by having comments

Reply:
comment will be removed. that's WIP

Reply:
This class is not part of this PR. comment is WIP